app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt

```diff
@@ -591,15 +599,42 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
         map!!.animateCamera(CameraUpdateFactory.newLatLngZoom(coordinates, zoomLevel))
     }
 
+    private fun toggleSaveButton() {
+        val saveButton : Button = requireActivity().findViewById(R.id.place_card_favorites_button)
+        if (saveButton.text.equals("Save")) {
+            saveButton.text = "Saved"
+            val redHeart : Drawable? = context?.resources?.getDrawable(R.drawable.ic_favorites_heart_red)
```

Comment:
Avoid using depreciated method as much as possible 😄 

Please replace: 

`val redHeart : Drawable? = context?.resources?.getDrawable(R.drawable.ic_favorites_heart_red)`

with:

`val redHeart = activity?.let { ContextCompat.getDrawable(it, R.drawable.ic_favorites_heart_red) }`